net/test/test-vdma.cpp

```diff
@@ -0,0 +1,239 @@
+#include "../../test/gtest.h"
+#include "../vdma/shm.cpp"
+#include <photon/photon.h>
+#include <photon/common/alog.h>
+#include <photon/thread/thread11.h>
+#include <photon/thread/workerpool.h>
+
+#include <vector>
+
+TEST(Buffer, encode_decode) {
+    std::string str;
+    photon::SharedMemoryBuffer::encode_to(str, 111, 222);
+    uint64_t idx, bufsz;
+    photon::SharedMemoryBuffer::decode_from(str, &idx, &bufsz);
```

Comment:
`&bufsz` is an uint64_t* while decode_from acquires size_t* for the third parameter.
In most cases it is same stuff but still, on macos with clang++, `size_t` refers `usigned long` while `uint64_t` is `unsigned long long`, these are different types of pointers, and will cause a compile failure.